fix(iota): fix table alignment, clippy

Thoralf-M opened the pull request:
# Description of change

Fix table alignment and applied clippy suggestions/ignored some graphql related ones. Updated wasm-bindgen because of this warning `the following packages contain code that will be rejected by a future version of Rust: wasm-bindgen v0.2.87`

## Links to any relevant issues

Partially addresses https://github.com/iotaledger/iota/issues/806

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

`cargo ci-clippy && cargo nextest run --profile ci -E 'rdeps(iota)' -p iota`

Thoralf-M committed:
fix(iota): fix table alignment, clippy

Thoralf-M committed:
More clippy

thibault-martinez reviewed (commented):
I'm getting another clippy warning on `crates/iota-types/src/effects/effects_v2.rs:459:9`, do you want to fix it?

Thoralf-M committed:
Even more clippy, update wasm-bindgen:
the following packages contain code that will be rejected by a future version of Rust: wasm-bindgen v0.2.87

Thoralf-M committed:
Use cfg(test) instead of allow(dead_code)

DaughterOfMars committed:
Merge branch 'develop' into dev-tools/CI-fixes
